src/Clock.tsx

```diff
@@ -21,8 +21,14 @@ export class Clock extends Component<Props, State> {
 
   componentDidMount() {
     this.timerCountId = window.setInterval(() => {
-      this.setState({ today: new Date() });
-      window.console.info(formatTime(this.state.today));
+      const today = new Date();
+
+      this.setState(
+        { today },
+        () => {
+          window.console.info(formatTime(this.state.today));
+        },
+      );
```

Comment:
Cool solution 🔥

You can even omit to create a variable cause the callback function will get an updated value.
```suggestion
      this.setState(
        { today: new Date() },
        () => {
          window.console.info(formatTime(this.state.today));
        },
      );
```